### PR TITLE
style: Update GridTable condensed styles

### PR DESCRIPTION
**Pull request by ksk5280**
This PR updates condensed styles to match the styles in our [custom product options superdrawer tables](https://www.figma.com/file/xF92q77bRnXXS7q7uRPDBK/21-Q2-%F0%9F%9B%A0Custom-Product-Options?node-id=536%3A869), per a suggestion from Stephen in [previous PR](https://github.com/homebound-team/internal-frontend/pull/1135#discussion_r666151928) and a follow up conversation with Dara to confirm that she was ok with this change.

### CODE REVIEW

**Review by stephenh — APPROVED** [review]
Sweet!

File: src/components/GridTable.test.tsx [hunk 1]
```diff
@@ -55,11 +55,6 @@ const nestedColumns: GridColumn<NestedRow>[] = [
 ];
 
 describe("GridTable", () => {
-  it("renders", async () => {
```
Comment (ksk5280) [inline src/components/GridTable.test.tsx:58]: @stephenh are you ok with deleting this snapshot test since we have stories?

Comment (stephenh) [reply src/components/GridTable.test.tsx:58]: Yep!

**Review by KoltonG — APPROVED** [review]
🍣 Thank you!!

**Pull request merged**